drift_dev/test/cli/make_migrations_test.dart

```diff
@@ -0,0 +1,193 @@
+import 'dart:io';
+
+import 'package:drift_dev/src/cli/cli.dart';
+import 'package:test/test.dart';
+import 'package:test_descriptor/test_descriptor.dart' as d;
+import 'package:path/path.dart' as p;
+
+import '../utils.dart';
+import 'utils.dart';
+
+void main() {
+  late TestDriftProject project;
+
+  group(
+    'make-migrations',
+    () {
+      tearDown(() async {
+        try {
+          await project.root.delete(recursive: true);
+        } catch (_) {}
+      });
+      test('default', () async {
+        project = await TestDriftProject.create([
+          d.dir('lib', [d.file('db.dart', _dbContent)]),
+          d.file('build.yaml', """
+targets:
+  \$default:
+    builders:
+      drift_dev:
+        options:
+          databases:
+            my_database: lib/db.dart""")
+        ]);
+        await project.runDriftCli(['make-migrations']);
+        expect(
+            d
+                .file('app/drift_schemas/my_database/drift_schema_v1.json')
+                .io
+                .existsSync(),
+            true);
+        // No other files should be created for 1st version
+        expect(d.file('app/test').io.existsSync(), false);
+        expect(d.file('app/lib/db.steps.dart').io.existsSync(), false);
+
+        // Change the db schema without bumping the version
+        File(p.join(project.root.path, 'lib/db.dart'))
+            .writeAsStringSync(_dbWithNewColumnWithoutVersionBump);
+        // Should throw an error
+        try {
+          await project.runDriftCli(['make-migrations']);
+          fail('Expected an error');
+        } catch (e) {
+          expect(e, isA<FatalToolError>());
+        }
+
+        // Change the db schema and bump the version
+        File(p.join(project.root.path, 'lib/db.dart'))
+            .writeAsStringSync(_dbWithNewColumnBump);
+        await project.runDriftCli(['make-migrations']);
+        expect(
```

Comment:
`package:test_descriptor` actually has a `validate` method that can validate entire directories (or at least multiple) files at once, which might make this test easier to understand. Can you check whether the validation part can be written [like in the example here](https://pub.dev/packages/test_descriptor#usage)?